pylibfuzzer/exec/runner.py

```diff
@@ -165,10 +182,24 @@ def run(self):
                 self.i += batchsize
                 self.input_generator.observe(results)
 
+                if self.create_dataset:
```

Comment:
I think its better if you do the whole dataset writing at this location and not split up the "open" and "write" in different parts of the code. Than you can also use a with statement instead of explicit open and close